### PR TITLE
Docs: Add top-level index page for KRM functions.

### PR DESCRIPTION
**Pull request by etefera**
Fixes https://github.com/GoogleContainerTools/kpt/issues/1066

image

image

### CODE REVIEW

File: site/content/en/guides/consumer/function/catalog/_index.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+title: "Function Catalog"
+linkTitle: "Function Catalog"
+type: docs
```
Comment (prachirp) [inline site/content/en/guides/consumer/function/catalog/_index.md:4]: Add weight field so this page appears above exporting workflows page

**Review by prachirp — CHANGES_REQUESTED** [review]
Can you update catalog pages with sops function and new examples too? I would consider sops as a transformer fn.

File: site/content/en/guides/consumer/function/catalog/_index.md (hunk 1 above)
Comment (etefera) [reply site/content/en/guides/consumer/function/catalog/_index.md:5]: Done.

**Review by prachirp — APPROVED** [review]
LGTM, will be a separate change to update catalog fns and examples

**Comment by frankfarzan** [comment]
I don't see changes to this page:

https://googlecontainertools.github.io/kpt/guides/consumer/function/

Which is the point of entry for catalog at this point.

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**